File: contracts/wasm/erc20/erc20.ts
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import * as wasmlib from "../wasmlib"
import * as sc from "./index";

export function funcApprove(ctx: wasmlib.ScFuncContext, f: sc.ApproveContext): void {
}

export function funcInit(ctx: wasmlib.ScFuncContext, f: sc.InitContext): void {
}

export function funcTransfer(ctx: wasmlib.ScFuncContext, f: sc.TransferContext): void {
}

export function funcTransferFrom(ctx: wasmlib.ScFuncContext, f: sc.TransferFromContext): void {
}

export function viewAllowance(ctx: wasmlib.ScViewContext, f: sc.AllowanceContext): void {
}

export function viewBalanceOf(ctx: wasmlib.ScViewContext, f: sc.BalanceOfContext): void {
}

export function viewTotalSupply(ctx: wasmlib.ScViewContext, f: sc.TotalSupplyContext): void {
}

File: contracts/wasm/erc20/src/erc20.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0


use wasmlib::*;

use crate::*;

pub fn func_approve(ctx: &ScFuncContext, f: &ApproveContext) {
}

pub fn func_init(ctx: &ScFuncContext, f: &InitContext) {
    if f.params.owner().exists() {
        f.state.owner().set_value(&f.params.owner().value());
        return;
    }
    f.state.owner().set_value(&ctx.request_sender());
}

pub fn func_transfer(ctx: &ScFuncContext, f: &TransferContext) {
}

pub fn func_transfer_from(ctx: &ScFuncContext, f: &TransferFromContext) {
}

pub fn view_allowance(ctx: &ScViewContext, f: &AllowanceContext) {
}

pub fn view_balance_of(ctx: &ScViewContext, f: &BalanceOfContext) {
}

pub fn view_total_supply(ctx: &ScViewContext, f: &TotalSupplyContext) {
}
